src/viam/sdk/common/proto_value.hpp

```diff
@@ -0,0 +1,324 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+namespace google {
+namespace protobuf {
+
+// Forward declaration of google::protobuf Value and Struct.
+// The class below is written so as to keep Value out of the ABI, and as such can be instantiated
+// with Value as an incomplete type.
+
+class Value;
+class Struct;
+}  // namespace protobuf
+}  // namespace google
+
+namespace viam {
+namespace sdk {
+
+namespace impl {
+
+struct move_may_throw {
+    move_may_throw() noexcept(false) = default;
+    move_may_throw& operator=(move_may_throw&&) noexcept(false) = default;
```

Comment:
Do you need move construction in here too?

Reply:
Oops yes that's a typo, first one should have been move construction